code/game/machinery/teleporter/console.dm

```diff
@@ -0,0 +1,187 @@
+			break
+	if (projector)
+		var/pad_dir
+		for (var/dir in GLOB.cardinal)
+			pad = locate() in get_step(projector, dir)
+			if (pad)
+				pad_dir = dir
+				break
+		if (pad)
+			projector.set_computer(src)
+			projector.set_dir(pad_dir)
+			pad.set_computer(src)
+			projector.queue_icon_update()
+			pad.queue_icon_update()
+
+
+/obj/machinery/computer/teleporter/proc/clear_projector()
+	if (!projector)
+		return
+	GLOB.destroyed_event.unregister(projector, src, /obj/machinery/computer/teleporter/proc/lost_projector)
+	projector = null
+	set_active(FALSE)
+
+
+/obj/machinery/computer/teleporter/proc/lost_projector()
+	audible_message(SPAN_WARNING("\The [src] buzzes, \"Projector missing.\""))
+	clear_projector()
+
+
+/obj/machinery/computer/teleporter/proc/set_projector(obj/machinery/tele_projector/_projector)
+	if (projector == _projector)
+		return
+	clear_projector()
+	projector = _projector
+	GLOB.destroyed_event.register(projector, src, /obj/machinery/computer/teleporter/proc/lost_projector)
+
+
+/obj/machinery/computer/teleporter/proc/clear_pad()
+	if (!pad)
+		return
+	GLOB.destroyed_event.unregister(pad, src, /obj/machinery/computer/teleporter/proc/lost_pad)
+	pad = null
+	set_active(FALSE)
+
+
+/obj/machinery/computer/teleporter/proc/lost_pad()
+	audible_message(SPAN_WARNING("\The [src] buzzes, \"Pad missing.\""))
+	clear_pad()
+
+
+/obj/machinery/computer/teleporter/proc/set_pad(obj/machinery/tele_pad/_pad)
+	if (pad == _pad)
+		return
+	clear_pad()
+	pad = _pad
+	GLOB.destroyed_event.register(pad, src, /obj/machinery/computer/teleporter/proc/lost_pad)
+
+
+/obj/machinery/computer/teleporter/proc/clear_target()
+	if (!target)
+		return
+	GLOB.destroyed_event.unregister(target, src, /obj/machinery/computer/teleporter/proc/lost_target)
+	target = null
+	set_active(FALSE)
+
+
+/obj/machinery/computer/teleporter/proc/lost_target()
+	audible_message(SPAN_WARNING("\The [src] buzzes, \"Target lost.\""))
+	clear_target()
+
+
+/obj/machinery/computer/teleporter/proc/set_target(atom/_target)
+	if (target == _target)
+		return
+	clear_target()
+	target = _target
+	GLOB.destroyed_event.register(target, src, /obj/machinery/computer/teleporter/proc/lost_target)
+
+
+/obj/machinery/computer/teleporter/proc/set_active(_active, notify)
+	var/effective = _active && target && projector && pad
+	if (active == effective)
+		return
+	active = effective
+	if (notify && effective)
+		if (active)
+			visible_message(SPAN_NOTICE("The teleporter sparks and hums to life."))
+		else
+			visible_message(SPAN_WARNING("The teleporter sputters and fails."))
+	if (projector)
+		projector.queue_icon_update()
+	if (pad)
+		pad.queue_icon_update()
+
+
+/obj/machinery/computer/teleporter/proc/get_targets()
+	var/list/ids = list()
+	var/list/result = list()
+	for (var/obj/item/radio/beacon/B)
+		if (QDELETED(B) || !B.functioning || !isPlayerLevel(B.z))
+			continue
+		var/area/A = get_area(B)
+		if (!A)
+			continue
+		result["[A.name] \[[++ids[A]]\]"] = B
```

Comment:
No being clever with prefix operators please. Split this out.

Reply:
I'm unsure what you mean with split this out or prefix operators.

Reply:
````
result["[A.name] \[[++ids[A]]\]"] = B
````
vs
````
var/id_val = ids[A]+1
ids[A] = id_val
result["[A.name] \[[id_val]\]"] = B
````